modules/core/keeper/keeper_test.go

```diff
@@ -0,0 +1,124 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	"github.com/stretchr/testify/suite"
+
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradekeeper "github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	ibchost "github.com/cosmos/ibc-go/v3/modules/core/24-host"
+	ibckeeper "github.com/cosmos/ibc-go/v3/modules/core/keeper"
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	"github.com/cosmos/ibc-go/v3/testing/simapp"
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	app           *simapp.SimApp
+	stakingKeeper clienttypes.StakingKeeper
+	upgradeKeeper clienttypes.UpgradeKeeper
+	scopedKeeper  capabilitykeeper.ScopedKeeper
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	coordinator := ibctesting.NewCoordinator(suite.T(), 1)
+	chainA := coordinator.GetChain(ibctesting.GetChainID(1))
+
+	suite.app = chainA.App.(*simapp.SimApp)
+
+	suite.stakingKeeper = suite.app.StakingKeeper
+	suite.upgradeKeeper = suite.app.UpgradeKeeper
+	suite.scopedKeeper = suite.app.ScopedIBCKeeper
+}
+
+func (suite *KeeperTestSuite) NewIBCKeeper() {
+	ibckeeper.NewKeeper(
+		suite.app.AppCodec(),
+		suite.app.GetKey(ibchost.StoreKey),
+		suite.app.GetSubspace(ibchost.ModuleName),
+		suite.stakingKeeper,
+		suite.upgradeKeeper,
+		suite.scopedKeeper,
+	)
```

Comment:
Could we place this inside the `suite.Run` func for `TestNewKeeper`. Typically in our tests, when using table tests, we use local variable for each of the inputs which will change

So
```go
func TestNewKeeper() {
var (
    stakingKeeper
    upgradeKeeper
    scopedKeeper
)

// table tests
func() { // malleate
    stakingKeeper = emptyStakingKeeper
}
...

suite.Run() {
    // setup
    stakingKeeper = suite.chainA.GetSimApp().StakingKeeper // set passing defaults
    // etc
    
    if tc.expPass {
        require.NotPanics(
            keeper.NewKeeper(stakingKeeper, ...)
        )
    } else {
        require.Panics(
            keeper.NewKeeper(stakingKeeper, ...)
         )
    }
}
```